### DISCUSSION TITLE
OncoKB Annotation Function

### DISCUSSION THREAD

**Discussion by carokos** [MSKCC-Epi-Bio/gnomeR#168] [Q&A]
Hi!

I'm trying to OncoKB annotate a large data set, and after it runs for a long time, I get this error:

`Error in py_call_impl(callable, dots$args, dots$keywords) : 
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
`

Any idea what might be happening or how to fix it?

Thanks!

**Comment by karissawhiting** [comment]
Hi @carokos, thanks for using discussions! 

It seems the API server is terminating your connection, but I'm not sure exactly why. How large is your data set? Could you please try taking a small sample from it and running through again to see if that works? If so, maybe it's a capacity issue.

**Reply by carokos** [reply]
It's pretty big... about 300,000 mutations.

From my most recent tests with smaller samples:
- 1,000 mutations: ran without errors
- 10,000 mutations: got the error after 5,000 mutations
- 100,000 mutations: got the error after 98,000 mutations

So it's really variable.  I've had it run with 10,000 and 100,000 mutations before.  Not sure what it's dependent on...

**Reply by karissawhiting** [reply]
Were the errors all the same? It sounds like it may be timing out after a certain period of time.

**Reply by carokos** [reply]
Yes! I get the same error each time.  I can try timing each run, but I think even the time is really variable (getting to 98k mutations takes much, much longer than getting to 5k mutations).  Do you mean timing out in terms of the run time, or the function/script trying to connect with the OncoKB API?

**Reply by karissawhiting** [reply]
The API might have limits on how fast /how often we can send requests, that we may be exceeding. Can you send me the data? I will test on my computer and add an issue with more info.  Thank you!

**Reply by carokos** [reply]
Makes sense.  Sending the data via email now. Thanks for looking into this!
